src/srcset.js

```diff
@@ -210,16 +210,19 @@ export class Srcset {
         // minimum values seen.
         sourceWidth = minWidth / 2;
       }
-      minWidth = Math.min(minWidth, sourceWidth);
-      // The calculation is slightly biased toward higher width by offsetting
-      // score by negative 0.2.
-      const score = Math.abs((sourceWidth - width) / width - 0.2);
-      if (score < minScore) {
-        minScore = score;
-        minIndex = i;
+      // First candidate width that's equal or higher than the requested width
+      // will stop the search.
+      if (sourceWidth >= width) {
+        // The right value is now between `i` and `i + 1` - select the one
+        // that is closer with a slight preference toward higher numbers.
+        const delta = sourceWidth - width;
+        const prevDelta = width - prevWidth * 0.9;
```

Comment:
The comment above explains it: we give a 10% preferences for a bigger width. This is a compromise between Chrome (that always selects the closest source) and Safari (that always selects the bigger source).